tmk_core/common/chibios/eeprom_stm32.c

```diff
@@ -14,185 +14,793 @@
  * Artur F.
  *
  * Modifications for QMK and STM32F303 by Yiancar
+ * Modifications to add flash wear leveling by Ilya Zhuravlev
+ * Modifications to increase flash density by Don Kjer
  */
 
 #include <stdio.h>
-#include <string.h>
+#include <stdbool.h>
 #include "eeprom_stm32.h"
-/*****************************************************************************
- * Allows to use the internal flash to store non volatile data. To initialize
- * the functionality use the EEPROM_Init() function. Be sure that by reprogramming
- * of the controller just affected pages will be deleted. In other case the non
- * volatile data will be lost.
- ******************************************************************************/
+#include "flash_stm32.h"
+#ifdef FLASH_STM32_MOCKED
```

Comment:
Not a massive fan of the whole `FLASH_STM32_MOCKED` thing. Not enough to block it from being merged, but enough that not a paradigm i would want to propagate and is going to have to be addressed at some point.

Reply:
I have attempted to reduce as much use of `FLASH_STM32_MOCKED` as I can.  However, there is still one use that I had trouble getting rid of:  https://github.com/qmk/qmk_firmware/pull/12567/files#diff-767cf2cd7a1a263022da956e08c956e0e8cddf8ec1eed149e33efea2081fd7b2R28

I am open to suggestions on a better approach